dm/worker/config.go

```diff
@@ -262,13 +271,19 @@ func (c *Config) adjustFlavor() error {
 	if err != nil {
 		return err
 	}
-	fromDB, err := applyNewBaseDB(clone.From)
+	from := clone.From
+	from.RawDBCfg = config.DefaultRawDBConfig().SetReadTimeout(flavorReadTimeout)
+	fromDB, err := applyNewBaseDB(from)
 	if err != nil {
 		return terror.WithScope(err, terror.ScopeUpstream)
 	}
 	defer fromDB.Close()
 
-	c.Flavor, err = utils.GetFlavor(fromDB.DB)
+	ctx, _ := context.WithTimeout(context.Background(), flavorGetTimeout)
```

Comment:
It's better to record `cancel` and add `defer cancel()`. comment from `context.WithTimeout`:
```
// Canceling this context releases resources associated with it, so code should
// call cancel as soon as the operations running in this Context **complete:**
```